src/scenic/syntax/ast.py

```diff
@@ -10,6 +10,57 @@ def __init__(self, *args: any, **kwargs: any) -> None:
 
 
 # special statements
+
+
+class ScenicTry(AST):
+    """Scenic AST node that represents try-catch statements with `interrupt when` clauses"""
```

Comment:
I'd call the class `TryInterrupt` and change the docstring accordingly, since these are called "try-interrupt" statements in the paper and documentation. It's true their syntax generalizes the Python `try` statement, but they're only valid in limited contexts (inside behaviors and `compose` blocks of modular scenarios), so I wouldn't consider them a form of the ordinary `try`.

Reply:
Fixed in e843c6c4af5948f148efb71414a917ad82b71f3c. I changed the name of the class to `TryInterrupt`.